R/filter-helpers.R

```diff
@@ -5,6 +5,7 @@
 #' @param dm A [`dm`] object
 #' @export
 cdm_nrow <- function(dm) {
+  check_no_filter(dm)
```

Comment:
Why is this important?

Reply:
I would think, if a user adds filter conditions and then calls `cdm_nrow()`, they are interested in the number rows after filtering. But since the function just counts the rows in the unfiltered state (using `cdm_get_tables()`) I figured it would make sense to add this function to those that don't work when a filter condition exists.

Reply:
Yeah, sure, we need `cdm_apply_filters()` to compute accurately.

Reply:
Sorry, got to ask once more to make sure: should we have in `cdm_nrow()` rather `check_no_filter(dm)` or `cdm_apply_filters(dm)`?